test/e2e/validators.go

```diff
@@ -300,11 +300,16 @@ func validateBuildDeletion(
 	}, time.Duration(30*getTimeoutMultiplier())*time.Second, 3*time.Second).ShouldNot(BeNil(), "Build is not deleted yet")
 
 	buildRunNsName := types.NamespacedName{Name: testBuildRun.Name, Namespace: namespace}
-	err = clientGet(buildRunNsName, testBuildRun)
 	if expectedDeletion {
-		Expect(apierrors.IsNotFound(err)).To(BeTrue(), "BuildRun was not deleted together with the Build")
+		// This asserts for Build that have the annotation to automaticall delete BuildRuns
+		// Poll until a BuildRun is deleted, by asserting if the object is not found.
+		// TODO: This test doest no belong here, but waiting for the integration test to be merged,
```

Comment:
Does that mean we have to block this until the integration test is fully processed?

Reply:
Good question. I dont think so. Once both PRS are merged, we can just do another pr to remove this e2e test, and ensure integration tests have the test spec for the ownerreferences. I need to keep this test here for now, otherwise nothings is testing this feature against a k8s cluster.